src/main/java/com/ldtteam/structurize/network/messages/ShowScanMessage.java

```diff
@@ -0,0 +1,103 @@
+package com.ldtteam.structurize.network.messages;
+
+import com.ldtteam.structurize.api.util.Log;
+import com.ldtteam.structurize.helpers.Settings;
+import com.ldtteam.structurize.util.ClientStructureWrapper;
+import io.netty.buffer.ByteBufInputStream;
+import io.netty.buffer.ByteBufOutputStream;
+import net.minecraft.core.BlockPos;
+import net.minecraft.nbt.NbtIo;
+import net.minecraft.nbt.CompoundTag;
+import net.minecraft.network.FriendlyByteBuf;
+import net.minecraft.util.Tuple;
+import net.minecraftforge.fml.LogicalSide;
+import net.minecraftforge.network.NetworkEvent;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+
+import java.io.IOException;
+import java.util.Optional;
+
+/**
+ * Handles ShowScanMessages.
+ */
+public class ShowScanMessage implements IMessage
+{
+    private final BlockPos from;
+    private final BlockPos to;
+    @Nullable private final BlockPos anchor;
+    private final String fileName;
+
+    /**
+     * Tell the client to update its scan region display.
+     *
+     * @param from The first corner.
+     * @param to The second corner.
+     * @param anchor The anchor, if any.
+     * @param fileName The scan filename.
+     */
+    public ShowScanMessage(@NotNull final BlockPos from,
+                           @NotNull final BlockPos to,
+                           @Nullable final BlockPos anchor,
+                           @NotNull final String fileName)
+    {
+        this.from = from;
+        this.to = to;
+        this.anchor = anchor;
+        this.fileName = fileName;
+    }
+
+    /**
+     * Deserialize.
+     */
+    public ShowScanMessage(final FriendlyByteBuf buf)
+    {
+        final FriendlyByteBuf buffer = new FriendlyByteBuf(buf);

```

Comment:
what?

Reply:
Fair point.  In my defense, I was following what this code does:

https://github.com/ldtteam/Structurize/blob/3d9952bb620656adee0c6a727e291722a5738cca/src/main/java/com/ldtteam/structurize/network/messages/SaveScanMessage.java#L32

But you're correct that this is silly, and I'll change my one.

Reply:
Super weird, might be some leftover from older times when duplicating was done for data security? w/e